rocketpy/Flight.py

```diff
@@ -2121,8 +2121,7 @@ def potentialEnergy(self):
         # Redefine totalMass time grid to allow for efficient Function algebra
         totalMass = deepcopy(self.rocket.totalMass)
         totalMass.setDiscreteBasedOnModel(self.z)
-        # TODO: change calculation method to account for variable gravity
-        potentialEnergy = totalMass * self.env.gravity.compose(self.z)
+        potentialEnergy = totalMass * self.env.gravity.compose(self.z) * self.z
```

Comment:
I am afraid this formula, $U_g (z) = mg(z)z$ is no longer valid for variable gravity. Instead, we should use $U_g (R) = \frac{GMm}{R}$ where $R$ is the rocket's center of mass position vector magnitude relative to Earth's center.

Reply:
So I implemented this in https://github.com/RocketPy-Team/RocketPy/pull/338/commits/90b2bdb1aa59d1128ed4c8955cd64de74297d791 with the following code:

```
        potentialEnergy = GM * totalMass / self.env.earthRadius * (1 / self.z - 1)
```

So what we are calculating here is the potential energy in relation to the sea level. All good now right?

Reply:
This is the result for Calisto:


![Figure 38](https://user-images.githubusercontent.com/69485049/233748163-313f00f7-0cde-4528-84ba-51c71ba5df3b.png)